dns/src/lib.rs

```diff
@@ -249,7 +249,7 @@ impl<'a, W5500: Udp> Response<'a, W5500> {
     /// Number of answers in the response.
     #[must_use]
     pub fn answer_count(&self) -> u16 {
-        self.header.ancount()
+        self.header.ancount() + self.header.arcount()
```

Comment:
`saturating_add` should be used, otherwise malformed headers with both RR's set to `u16::MAX` will panic with overflow checks enabled.